cli/tests/test_split_command.rs

```diff
@@ -675,3 +680,199 @@ fn test_split_interactive_with_paths() {
+    default: qpvuntsm d12cb1c0 first-commit
+    second: qpvuntsm d12cb1c0 first-commit
+    "###);
+    // Check the working-copy commit in each workspace in the log output. The "@"
+    // node in the graph indicates the current workspace's working-copy commit.
+    insta::assert_snapshot!(get_workspace_log_output(&test_env, &main_path), @r###"
+    @  qpvuntsmwlqt default@ second@ first-commit
+    ◆  zzzzzzzzzzzz
+    "###);
+    insta::assert_snapshot!(get_workspace_log_output(&test_env, &secondary_path), @r###"
+    @  qpvuntsmwlqt default@ second@ first-commit
+    ◆  zzzzzzzzzzzz
+    "###);
+
+    // Do the split in the default workspace.
+    std::fs::write(
+        test_env.set_up_fake_editor(),
+        ["", "next invocation\n", "write\nsecond-commit"].join("\0"),
+    )
+    .unwrap();
+    test_env.jj_cmd_ok(&main_path, &["split", "file2"]);
+    // The working copy for both workspaces will be the second split commit.
+    let stdout = test_env.jj_cmd_success(&main_path, &["workspace", "list"]);
+    insta::assert_snapshot!(stdout, @r###"
+    default: vruxwmqv 97490ec0 second-commit
+    second: vruxwmqv 97490ec0 second-commit
+    "###);
+    insta::assert_snapshot!(get_workspace_log_output(&test_env, &main_path), @r###"
+    @  vruxwmqvtpmx default@ second@ second-commit
+    ○  qpvuntsmwlqt first-commit
+    ◆  zzzzzzzzzzzz
+    "###);
+    insta::assert_snapshot!(get_workspace_log_output(&test_env, &secondary_path), @r###"
+    @  vruxwmqvtpmx default@ second@ second-commit
+    ○  qpvuntsmwlqt first-commit
+    ◆  zzzzzzzzzzzz
+    "###);
+
+    // Test again with a --parallel split.
+    test_env.jj_cmd_ok(&main_path, &["undo"]);
+    std::fs::write(
+        test_env.set_up_fake_editor(),
+        ["", "next invocation\n", "write\nsecond-commit"].join("\0"),
+    )
+    .unwrap();
+    test_env.jj_cmd_ok(&main_path, &["split", "file2", "--parallel"]);
+    insta::assert_snapshot!(stdout, @r###"
+    default: vruxwmqv 97490ec0 second-commit
+    second: vruxwmqv 97490ec0 second-commit
+    "###);
+    insta::assert_snapshot!(get_workspace_log_output(&test_env, &main_path), @r###"
+    @  wqnwkozpkust default@ second@ second-commit
+    │ ○  qpvuntsmwlqt first-commit
+    ├─╯
+    ◆  zzzzzzzzzzzz
+    "###);
+    let stderr = test_env.jj_cmd_failure(&secondary_path, &["log"]);
+    // The working copy for the second workspace will be stale.
+    insta::assert_snapshot!(stderr, @r###"
+    Error: The working copy is stale (not updated since operation ff670112b6f3).
+    Hint: Run `jj workspace update-stale` to update it.
+    See https://jj-vcs.github.io/jj/latest/working-copy/#stale-working-copy for more information.
+    "###);
+    test_env.jj_cmd_ok(&secondary_path, &["workspace", "update-stale"]);
+    insta::assert_snapshot!(get_workspace_log_output(&test_env, &secondary_path), @r###"
+    @  wqnwkozpkust default@ second@ second-commit
+    │ ○  qpvuntsmwlqt first-commit
+    ├─╯
+    ◆  zzzzzzzzzzzz
+    "###);
```

Comment:
nit: this should already be covered by `test_workspaces.rs`

Reply:
There aren't any tests that split a commit in `test_workspaces.rs`, but there are tests for the working copy becoming stale after a commit is rewritten. Maybe that's good enough. I'll delete this.